packages/nextra-theme-blog/src/blog-context.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { useContext } from 'react'
+import { createContext, PropsWithChildren } from 'react'
+import { LayoutProps } from './types'
+
+const BlogContext = createContext<LayoutProps | null>(null)
+
+export const BlogProvider: React.FC<PropsWithChildren<LayoutProps>> = ({
+  config,
+  children,
+  opts
+}) => {
+  return (
+    <BlogContext.Provider value={{ config, opts }}>
```

Comment:
Value should be memoized otherwise every rerender - new object

Reply:
It's a top level component. It probably wont rerender often.

Reply:
it's will be removed https://github.com/shuding/nextra/pull/602#discussion_r932204813